src/main/scala/com/ubirch/filter/services/kafka/DefaultFilterService.scala

```diff
@@ -238,37 +275,36 @@ abstract class AbstractFilterService(cache: Cache, finder: Finder, config: Confi
 
   def forwardUPP(data: ProcessingData): Future[Option[RecordMetadata]] = {
 
-    val result1 = addToFilterCache(data)
-
-    val result2 = if (data.pm.getHint == UPP_TYPE_DELETE || data.pm.getHint == UPP_TYPE_DISABLE)
-      deleteFromVerificationCache(data)
-    else if (data.pm.getHint != UPP_TYPE_ENABLE)
-      addToVerificationCache(data)
-    else Future.successful(())
-
-    val result3 = send(data.cr.toProducerRecord(topic = producerForwardTopic))
-      .recoverWith { case _ => send(data.cr.toProducerRecord(topic = producerForwardTopic)) }
-      .recoverWith { case ex =>
-        pauseKafkaConsumption(s"kafka error, not able to publish  ${data.cr.requestIdHeader().orNull} to $producerForwardTopic", data.cr, ex, FiniteDuration(2, SECONDS))
-      }
-    result3.onComplete {
-      case Success(_) =>
-        val hardwareId = data.cr.findHeader(HARDWARE_ID_HEADER_KEY).orNull
-        val requestId = data.cr.requestIdHeader().orNull
-        logger.info(s"Successfully forwarded msg from $hardwareId with requestId: $requestId", v("requestId", requestId), v("hardwareId", hardwareId))
-      case _ =>
-    }
-
     for {
-      _ <- result1
-      _ <- result2
-      recordMetaData <- result3
+      _ <- addToFilterCache(data)
```

Comment:
thank you for fixing the bad variable names I added

One thing:
In this code, these three procedures are executed sequentially.
Is it on purpose?
In the previous code, these are executed in parallel.

Reply:
this is the trick of Future.
this reference maybe helpful to understand it
https://stackoverflow.com/questions/43296040/why-arent-my-scala-futures-executing-in-parallel